lab_4_retrieval_w_clustering/main.py

```diff
@@ -81,6 +103,18 @@ def _calculate_bm25(self, tokenized_document: list[str]) -> Vector:
         Returns:
             Vector: BM25 vector for document.
         """
+        if not tokenized_document:
+            raise ValueError
+        vector = [0.0 for _ in range(len(self._vocabulary))]
+        document_len = len(tokenized_document)
+        idf_value = self._idf_values
+        dict_bm25 = calculate_bm25(self._vocabulary, tokenized_document,
+                                   idf_value, avg_doc_len=self._avg_doc_len, doc_len=document_len)
+        if dict_bm25 is None:
+            raise ValueError('_calculate_bm25 returned None, please check implemention')
+        for i, word in self._token2ind:
```

Comment:
why are you iterating over token2ind?

Reply:
i can't understand, where i need to use token2ind...